[Bug]: `TestRowCountBasedBalancerSuite/TestBalance/balance_unstable_view` unit test is not stable

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 9d54d6f
- Deployment mode(standalone or cluster): both
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```


### Current Behavior

 `TestRowCountBasedBalancerSuite/TestBalance/balance_unstable_view` unit test is not stable

### Expected Behavior

unit test shall pass stably

### Steps To Reproduce

_No response_

### Milvus Log

```
2023-12-11T15:37:14.1689385Z === RUN   TestRowCountBasedBalancerSuite/TestBalance/balance_unstable_view
2023-12-11T15:37:14.1709505Z [2023/12/11 15:37:14.170 +00:00] [DEBUG] [meta/target_manager.go:178] ["start to pull next targets for collection"] [collectionID=1] [chosenPartitionIDs="[1]"]
2023-12-11T15:37:14.1713332Z [2023/12/11 15:37:14.170 +00:00] [DEBUG] [meta/target_manager.go:119] ["finish to update next targets for collection"] [collectionID=1] [PartitionIDs="[1]"] [segments="[2,3,4,5,1]"] [channels="[]"]
2023-12-11T15:37:14.1715397Z [2023/12/11 15:37:14.171 +00:00] [DEBUG] [meta/target_manager.go:71] ["start to update current target for collection"] [collectionID=1]
2023-12-11T15:37:14.1717892Z [2023/12/11 15:37:14.171 +00:00] [DEBUG] [meta/target_manager.go:81] ["finish to update current target for collection"] [collectionID=1] [segments="[1,2,3,4,5]"] [channels="[]"] [version=1702309034170523840]
2023-12-11T15:37:14.1721186Z [2023/12/11 15:37:14.171 +00:00] [DEBUG] [meta/target_manager.go:178] ["start to pull next targets for collection"] [collectionID=1] [chosenPartitionIDs="[1]"]
2023-12-11T15:37:14.1724398Z [2023/12/11 15:37:14.171 +00:00] [DEBUG] [meta/target_manager.go:119] ["finish to update next targets for collection"] [collectionID=1] [PartitionIDs="[1]"] [segments="[1,2,3,4,5]"] [channels="[]"]
2023-12-11T15:37:14.1735286Z     rowcount_based_balancer_test.go:388: 
2023-12-11T15:37:14.1738242Z         	Error Trace:	/go/src/github.com/milvus-io/milvus/internal/querycoordv2/balance/rowcount_based_balancer_test.go:388
2023-12-11T15:37:14.1740078Z         	            				/go/pkg/mod/github.com/stretchr/testify@v1.8.4/suite/suite.go:112
2023-12-11T15:37:14.1741154Z         	Error:      	elements differ
2023-12-11T15:37:14.1741893Z         	            	
2023-12-11T15:37:14.1744578Z         	            	extra elements in list A:
2023-12-11T15:37:14.1745077Z         	            	([]interface {}) (len=1) {
2023-12-11T15:37:14.1747218Z         	            	 (balance.ChannelAssignPlan) {
2023-12-11T15:37:14.1747740Z         	            	  Channel: (*meta.DmChannel)({
2023-12-11T15:37:14.1748292Z         	            	   VchannelInfo: (*datapb.VchannelInfo)({
2023-12-11T15:37:14.1749030Z         	            	    CollectionID: (int64) 1,
2023-12-11T15:37:14.1749543Z         	            	    ChannelName: (string) (len=2) "v2",
2023-12-11T15:37:14.1750358Z         	            	    SeekPosition: (*msgpb.MsgPosition)(<nil>),
2023-12-11T15:37:14.1751242Z         	            	    UnflushedSegments: ([]*datapb.SegmentInfo) <nil>,
2023-12-11T15:37:14.1752084Z         	            	    FlushedSegments: ([]*datapb.SegmentInfo) <nil>,
2023-12-11T15:37:14.1752691Z         	            	    DroppedSegments: ([]*datapb.SegmentInfo) <nil>,
2023-12-11T15:37:14.1753239Z         	            	    UnflushedSegmentIds: ([]int64) <nil>,
2023-12-11T15:37:14.1753694Z         	            	    FlushedSegmentIds: ([]int64) <nil>,
2023-12-11T15:37:14.1754495Z         	            	    DroppedSegmentIds: ([]int64) <nil>,
2023-12-11T15:37:14.1755159Z         	            	    IndexedSegmentIds: ([]int64) <nil>,
2023-12-11T15:37:14.1755844Z         	            	    IndexedSegments: ([]*datapb.SegmentInfo) <nil>,
2023-12-11T15:37:14.1756431Z         	            	    XXX_NoUnkeyedLiteral: (struct {}) {
2023-12-11T15:37:14.1757080Z         	            	    },
2023-12-11T15:37:14.1757624Z         	            	    XXX_unrecognized: ([]uint8) <nil>,
2023-12-11T15:37:14.1758287Z         	            	    XXX_sizecache: (int32) 0
2023-12-11T15:37:14.1758877Z         	            	   }),
2023-12-11T15:37:14.1759373Z         	            	   Node: (int64) 1,
2023-12-11T15:37:14.1759892Z         	            	   Version: (int64) 0
2023-12-11T15:37:14.1760486Z         	            	  }),
2023-12-11T15:37:14.1760937Z         	            	  ReplicaID: (int64) 1,
2023-12-11T15:37:14.1761397Z         	            	  From: (int64) 1,
2023-12-11T15:37:14.1761858Z         	            	  To: (int64) 2
2023-12-11T15:37:14.1762271Z         	            	 }
2023-12-11T15:37:14.1762526Z         	            	}
2023-12-11T15:37:14.1762951Z         	            	
2023-12-11T15:37:14.1763364Z         	            	
2023-12-11T15:37:14.1763838Z         	            	extra elements in list B:
2023-12-11T15:37:14.1764457Z         	            	([]interface {}) (len=1) {
2023-12-11T15:37:14.1765110Z         	            	 (balance.ChannelAssignPlan) {
2023-12-11T15:37:14.1765765Z         	            	  Channel: (*meta.DmChannel)({
2023-12-11T15:37:14.1766462Z         	            	   VchannelInfo: (*datapb.VchannelInfo)({
2023-12-11T15:37:14.1767152Z         	            	    CollectionID: (int64) 1,
2023-12-11T15:37:14.1767996Z         	            	    ChannelName: (string) (len=2) "v2",
2023-12-11T15:37:14.1769114Z         	            	    SeekPosition: (*msgpb.MsgPosition)(<nil>),
2023-12-11T15:37:14.1769919Z         	            	    UnflushedSegments: ([]*datapb.SegmentInfo) <nil>,
2023-12-11T15:37:14.1770887Z         	            	    FlushedSegments: ([]*datapb.SegmentInfo) <nil>,
2023-12-11T15:37:14.1771692Z         	            	    DroppedSegments: ([]*datapb.SegmentInfo) <nil>,
2023-12-11T15:37:14.1772636Z         	            	    UnflushedSegmentIds: ([]int64) <nil>,
2023-12-11T15:37:14.1773383Z         	            	    FlushedSegmentIds: ([]int64) <nil>,
2023-12-11T15:37:14.1774192Z         	            	    DroppedSegmentIds: ([]int64) <nil>,
2023-12-11T15:37:14.1774903Z         	            	    IndexedSegmentIds: ([]int64) <nil>,
2023-12-11T15:37:14.1775935Z         	            	    IndexedSegments: ([]*datapb.SegmentInfo) <nil>,
2023-12-11T15:37:14.1776869Z         	            	    XXX_NoUnkeyedLiteral: (struct {}) {
2023-12-11T15:37:14.1777496Z         	            	    },
2023-12-11T15:37:14.1778042Z         	            	    XXX_unrecognized: ([]uint8) <nil>,
2023-12-11T15:37:14.1778703Z         	            	    XXX_sizecache: (int32) 0
2023-12-11T15:37:14.1779415Z         	            	   }),
2023-12-11T15:37:14.1779883Z         	            	   Node: (int64) 1,
2023-12-11T15:37:14.1780439Z         	            	   Version: (int64) 0
2023-12-11T15:37:14.1780981Z         	            	  }),
2023-12-11T15:37:14.1781478Z         	            	  ReplicaID: (int64) 1,
2023-12-11T15:37:14.1782168Z         	            	  From: (int64) 1,
2023-12-11T15:37:14.1782723Z         	            	  To: (int64) 3
2023-12-11T15:37:14.1783225Z         	            	 }
2023-12-11T15:37:14.1783639Z         	            	}
2023-12-11T15:37:14.1784063Z         	            	
2023-12-11T15:37:14.1784473Z         	            	
2023-12-11T15:37:14.1784875Z         	            	listA:
2023-12-11T15:37:14.1785434Z         	            	([]balance.ChannelAssignPlan) (len=1) {
2023-12-11T15:37:14.1786501Z         	            	 (balance.ChannelAssignPlan) {
2023-12-11T15:37:14.1787177Z         	            	  Channel: (*meta.DmChannel)({
2023-12-11T15:37:14.1787888Z         	            	   VchannelInfo: (*datapb.VchannelInfo)({
2023-12-11T15:37:14.1788596Z         	            	    CollectionID: (int64) 1,
2023-12-11T15:37:14.1789263Z         	            	    ChannelName: (string) (len=2) "v2",
2023-12-11T15:37:14.1789979Z         	            	    SeekPosition: (*msgpb.MsgPosition)(<nil>),
2023-12-11T15:37:14.1790774Z         	            	    UnflushedSegments: ([]*datapb.SegmentInfo) <nil>,
2023-12-11T15:37:14.1791606Z         	            	    FlushedSegments: ([]*datapb.SegmentInfo) <nil>,
2023-12-11T15:37:14.1792480Z         	            	    DroppedSegments: ([]*datapb.SegmentInfo) <nil>,
2023-12-11T15:37:14.1793258Z         	            	    UnflushedSegmentIds: ([]int64) <nil>,
2023-12-11T15:37:14.1793882Z         	            	    FlushedSegmentIds: ([]int64) <nil>,
2023-12-11T15:37:14.1794287Z         	            	    DroppedSegmentIds: ([]int64) <nil>,
2023-12-11T15:37:14.1794692Z         	            	    IndexedSegmentIds: ([]int64) <nil>,
2023-12-11T15:37:14.1795153Z         	            	    IndexedSegments: ([]*datapb.SegmentInfo) <nil>,
2023-12-11T15:37:14.1795928Z         	            	    XXX_NoUnkeyedLiteral: (struct {}) {
2023-12-11T15:37:14.1796549Z         	            	    },
2023-12-11T15:37:14.1796890Z         	            	    XXX_unrecognized: ([]uint8) <nil>,
2023-12-11T15:37:14.1797274Z         	            	    XXX_sizecache: (int32) 0
2023-12-11T15:37:14.1797611Z         	            	   }),
2023-12-11T15:37:14.1797885Z         	            	   Node: (int64) 1,
2023-12-11T15:37:14.1798204Z         	            	   Version: (int64) 0
2023-12-11T15:37:14.1798516Z         	            	  }),
2023-12-11T15:37:14.1798801Z         	            	  ReplicaID: (int64) 1,
2023-12-11T15:37:14.1799129Z         	            	  From: (int64) 1,
2023-12-11T15:37:14.1799442Z         	            	  To: (int64) 2
2023-12-11T15:37:14.1799732Z         	            	 }
2023-12-11T15:37:14.1799969Z         	            	}
2023-12-11T15:37:14.1800204Z         	            	
2023-12-11T15:37:14.1800432Z         	            	
2023-12-11T15:37:14.1800653Z         	            	listB:
2023-12-11T15:37:14.1800975Z         	            	([]balance.ChannelAssignPlan) (len=1) {
2023-12-11T15:37:14.1801388Z         	            	 (balance.ChannelAssignPlan) {
2023-12-11T15:37:14.1801878Z         	            	  Channel: (*meta.DmChannel)({
2023-12-11T15:37:14.1802341Z         	            	   VchannelInfo: (*datapb.VchannelInfo)({
2023-12-11T15:37:14.1802742Z         	            	    CollectionID: (int64) 1,
2023-12-11T15:37:14.1803127Z         	            	    ChannelName: (string) (len=2) "v2",
2023-12-11T15:37:14.1803556Z         	            	    SeekPosition: (*msgpb.MsgPosition)(<nil>),
2023-12-11T15:37:14.1804022Z         	            	    UnflushedSegments: ([]*datapb.SegmentInfo) <nil>,
2023-12-11T15:37:14.1804498Z         	            	    FlushedSegments: ([]*datapb.SegmentInfo) <nil>,
2023-12-11T15:37:14.1804968Z         	            	    DroppedSegments: ([]*datapb.SegmentInfo) <nil>,
2023-12-11T15:37:14.1805476Z         	            	    UnflushedSegmentIds: ([]int64) <nil>,
2023-12-11T15:37:14.1805890Z         	            	    FlushedSegmentIds: ([]int64) <nil>,
2023-12-11T15:37:14.1806288Z         	            	    DroppedSegmentIds: ([]int64) <nil>,
2023-12-11T15:37:14.1806740Z         	            	    IndexedSegmentIds: ([]int64) <nil>,
2023-12-11T15:37:14.1807179Z         	            	    IndexedSegments: ([]*datapb.SegmentInfo) <nil>,
2023-12-11T15:37:14.1807609Z         	            	    XXX_NoUnkeyedLiteral: (struct {}) {
2023-12-11T15:37:14.1807960Z         	            	    },
2023-12-11T15:37:14.1808272Z         	            	    XXX_unrecognized: ([]uint8) <nil>,
2023-12-11T15:37:14.1808651Z         	            	    XXX_sizecache: (int32) 0
2023-12-11T15:37:14.1808979Z         	            	   }),
2023-12-11T15:37:14.1809258Z         	            	   Node: (int64) 1,
2023-12-11T15:37:14.1809579Z         	            	   Version: (int64) 0
2023-12-11T15:37:14.1809898Z         	            	  }),
2023-12-11T15:37:14.1810431Z         	            	  ReplicaID: (int64) 1,
2023-12-11T15:37:14.1810776Z         	            	  From: (int64) 1,
2023-12-11T15:37:14.1811096Z         	            	  To: (int64) 3
2023-12-11T15:37:14.1811388Z         	            	 }
2023-12-11T15:37:14.1811631Z         	            	}
2023-12-11T15:37:14.1812055Z         	Test:       	TestRowCountBasedBalancerSuite/TestBalance/balance_unstable_view
2023-12-11T15:37:14.1813998Z [2023/12/11 15:37:14.173 +00:00] [DEBUG] [etcd/etcd_kv.go:66] ["etcd kv closed"] [path=by-dev/meta-test-querycoord8035416973958879266/meta-test-querycoord4213157934141147846/meta-test-querycoord2506709404761328192/meta]
...
2023-12-11T15:37:14.2176030Z --- FAIL: TestRowCountBasedBalancerSuite (0.13s)
2023-12-11T15:37:14.2177236Z     --- PASS: TestRowCountBasedBalancerSuite/TestAssignSegment (0.00s)
2023-12-11T15:37:14.2178627Z         --- PASS: TestRowCountBasedBalancerSuite/TestAssignSegment/test_normal_assignment (0.00s)
2023-12-11T15:37:14.2180080Z     --- PASS: TestRowCountBasedBalancerSuite/TestAssignSegmentWithGrowing (0.00s)
2023-12-11T15:37:14.2181301Z     --- FAIL: TestRowCountBasedBalancerSuite/TestBalance (0.06s)
2023-12-11T15:37:14.2182497Z         --- PASS: TestRowCountBasedBalancerSuite/TestBalance/normal_balance (0.01s)
2023-12-11T15:37:14.2184008Z         --- PASS: TestRowCountBasedBalancerSuite/TestBalance/skip_balance_for_redundant_segment (0.01s)
2023-12-11T15:37:14.2185698Z         --- PASS: TestRowCountBasedBalancerSuite/TestBalance/balance_won't_trigger (0.01s)
2023-12-11T15:37:14.2187128Z         --- PASS: TestRowCountBasedBalancerSuite/TestBalance/all_stopping_balance (0.01s)
2023-12-11T15:37:14.2188671Z         --- PASS: TestRowCountBasedBalancerSuite/TestBalance/part_stopping_balance (0.01s)
2023-12-11T15:37:14.2190090Z         --- PASS: TestRowCountBasedBalancerSuite/TestBalance/balance_channel (0.00s)
2023-12-11T15:37:14.2191437Z         --- PASS: TestRowCountBasedBalancerSuite/TestBalance/unbalance_stable_view (0.00s)
2023-12-11T15:37:14.2192838Z         --- FAIL: TestRowCountBasedBalancerSuite/TestBalance/balance_unstable_view (0.01s)
```

### Anything else?

_No response_

--- a/internal/querycoordv2/balance/rowcount_based_balancer_test.go
+++ b/internal/querycoordv2/balance/rowcount_based_balancer_test.go
@@ -148,6 +148,7 @@ func (suite *RowCountBasedBalancerTestSuite) TestBalance() {
 		distributionChannels map[int64][]*meta.DmChannel
 		expectPlans          []SegmentAssignPlan
 		expectChannelPlans   []ChannelAssignPlan
+		multiple             bool
 	}{
 		{
 			name:        "normal balance",
@@ -307,7 +308,9 @@ func (suite *RowCountBasedBalancerTestSuite) TestBalance() {
 			expectPlans: []SegmentAssignPlan{},
 			expectChannelPlans: []ChannelAssignPlan{
 				{Channel: &meta.DmChannel{VchannelInfo: &datapb.VchannelInfo{CollectionID: 1, ChannelName: "v2"}, Node: 1}, From: 1, To: 2, ReplicaID: 1},
+				{Channel: &meta.DmChannel{VchannelInfo: &datapb.VchannelInfo{CollectionID: 1, ChannelName: "v2"}, Node: 1}, From: 1, To: 3, ReplicaID: 1},
 			},
+			multiple: true,
 		},
 		{
 			name:            "already balanced",
@@ -385,8 +388,13 @@ func (suite *RowCountBasedBalancerTestSuite) TestBalance() {
 			}
 
 			segmentPlans, channelPlans := suite.getCollectionBalancePlans(balancer, 1)
-			suite.ElementsMatch(c.expectChannelPlans, channelPlans)
-			suite.ElementsMatch(c.expectPlans, segmentPlans)
+			if !c.multiple {
+				suite.ElementsMatch(c.expectChannelPlans, channelPlans)
+				suite.ElementsMatch(c.expectPlans, segmentPlans)
+			} else {
+				suite.Subset(c.expectPlans, segmentPlans)
+				suite.Subset(c.expectChannelPlans, channelPlans)
+			}
 
 			// clear distribution
 			for node := range c.distributions {

